packages/components/addon/components/hds/text/body.js

```diff
@@ -0,0 +1,68 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+// notice: only some combinations of size + font-weight are allowed (per design specs)
+// see: https://www.figma.com/file/oQsMzMMnynfPWpMEt91OpH/HDS-Product---Foundations?node-id=1262%3A9192
+
+export const DEFAULT_SIZE = '200';
+export const AVAILABLE_SIZES = [300, 200, 100];
+
+export const DEFAULT_WEIGHT = 'regular';
+export const AVAILABLE_WEIGHTS_PER_SIZE = {
+  300: ['regular', 'medium', 'semibold'],
+  200: ['regular', 'medium', 'semibold'],
+  100: ['regular', 'medium', 'semibold'],
+};
+
+export default class HdsTextBodyComponent extends Component {
+  /**
+   * Sets the "size" for the text
+   * Accepted values: see AVAILABLE_SIZES
+   *
+   * @type {string}
+   *
+   * @param size
+   */
+  get size() {
+    let { size = DEFAULT_SIZE } = this.args;
+
+    // let's be a bit forgiving with the consumers
```

Comment:
Nice! This is similar to what I did for the `@spacing` arg for my `Gutters` layout component spike: https://github.com/hashicorp/design-system/pull/538/files#r955427462